src/templates/websocket.js

```diff
@@ -0,0 +1,87 @@
+(async (addr, hard) => {
+    addr = `ws://${addr}/live-server-ws`;
+    const sleep = (x) => new Promise(r => setTimeout(r, x));
+    const preload = async (url, requireSuccess) => {
+        const resp = await fetch(url, {cache: "reload"}); // reset cache
+        if (requireSuccess && (!resp.ok || resp.status !== 200)) { throw new Error(); }
+    }
+    /** Reset cache in link.href and strip scripts */
+    const preloadNode = (n, ps) => {
+        if (n.tagName === "SCRIPT" && n.src) { ps.push(preload(n.src, false)); return; }
+        if (n.tagName === "LINK" && n.href) { ps.push(preload(n.href, false)); return; }
+        let c = n.firstChild; while (c) { let nc = c.nextSibling; preloadNode(c, ps); c = nc; }
+    };
+    let reloading = false;
+    let scheduled = false;
+    async function reload() {
+        if(reloading) {
+            scheduled = true;
+            return;
+        }
+        let ifr;
+        let interval = 0;
+        reloading = true;
+        while(true) {
+            scheduled = false;
+            try {
+                const url = location.origin + location.pathname;
+                const ps = [];
+                preloadNode(document.head, ps);
+                preloadNode(document.body, ps);
+                await Promise.allSettled(ps);
+                await new Promise((resolve) => {
+                    ifr = document.createElement("iframe");
+                    ifr.src = url + "?reload";
+                    ifr.style.display = "none";
+                    ifr.onload = resolve;
+                    document.body.appendChild(ifr);
+                });
+                const meta = ifr.contentDocument.head.lastChild;
+                if (meta.tagName !== "META" || meta.name !== "live-server" || meta.content !== "reload") {
```

Comment:
Is the meta tag necessary? I am not sure what it is used for.

Reply:
See my other comment